substrate/frame/revive/src/vm/evm/instructions/macros.rs

```diff
@@ -122,51 +122,19 @@ macro_rules! gas_or_fail_legacy {
 	};
 }
 
-use crate::vm::Ext;
-use revm::interpreter::gas::{MemoryExtensionResult, MemoryGas};
-
-/// Adapted from
-/// https://docs.rs/revm/latest/revm/interpreter/struct.Gas.html#method.record_memory_expansion
-pub fn record_memory_expansion<E: Ext>(
-	memory: &mut MemoryGas,
-	_ext: &mut E,
-	new_len: usize,
-) -> MemoryExtensionResult {
-	let Some(_additional_cost) = memory.record_new_len(new_len) else {
-		return MemoryExtensionResult::Same;
-	};
-
-	// TODO: Commented this for now so that extending memory will not cost any gas.
-	// if ext.gas_meter_mut().charge_evm_gas(additional_cost).is_err() {
-	// 	return MemoryExtensionResult::OutOfGas;
-	// }
-
-	MemoryExtensionResult::Extended
-}
-
-/// Resizes the interpreterreter memory if necessary. Fails the instruction if the memory or gas
-/// limit is exceeded.
+/// Ensures that memory access at `offset..offset+len` is valid.
+/// Revert if the memory is out of bounds.
 #[macro_export]
-macro_rules! resize_memory {
+macro_rules! check_memory_bounds {
```

Comment:
Nit: This has to be a macro?

Reply:
right now yes, because it returns on Error and all the REVM function do not return Result so you can't use `?` to propagate errors